prettifier_test.go

```diff
@@ -21,6 +21,30 @@ func TestPrettifyError(t *testing.T) {
 	f(`invalid query`)
 }
 
+func TestPrettifyOkParseError(t *testing.T) {
+	f := func(s string, wantErr string) {
+		t.Helper()
+
+		// check prettify ok
+		// it should not check query for semantic errors
+		result, err := Prettify(s)
+		if err != nil {
+			t.Fatalf("unexpected error when prettifying: %q: %s", s, err)
+		}
+
+		// Verify that the prettified result is failed to parse due to semantic error
+		_, err = Parse(result)
+		if err == nil {
+			t.Fatalf("expecting non-nil error")
+		}
+		if err.Error() != wantErr {
+			t.Fatalf("unexpected error string:\ngot:\n%q\nwant:\n%q", err, wantErr)
+		}
+	}
+
+	f(`sum(rate{a="b"}[BAZ])`, `cannot expand WITH expressions: cannot parse window for rate{a="b"}: cannot find WITH template for "BAZ"`)
```

Comment:
weird to have only one test case, it could have been just function code and that's all

Reply:
Added 1 more test case.

 This function has clear description for test case and do not complicates test logic for TestPrettifyError function.